### PR TITLE
Bug 1908971: Gherkin script for pipeline tech debt 4.7

### PR DESCRIPTION
**Pull request by sanketpathak**
Epic: https://issues.redhat.com/browse/ODC-4392
Story: https://issues.redhat.com/browse/ODC-4849

### CODE REVIEW

File: frontend/packages/dev-console/integration-tests/features/pipelines/create-from-builder-page.feature [hunk 1]
```diff
@@ -87,10 +87,27 @@ Scenario: Add Paramters to the pipeline in pipeline builder page : P-03-TC04
...
+    And user adds "name: git-clone" and "kind: ClusterTask" under taskRef in tasks section 
+    And user clicks Pipeline Builder radio button 
+    Then user can see "git clone" tasks present in the Pipeline builder page
+
+
 @regression, @manual
 Scenario: Create the pipeline from yaml editor : P-07-TC01
     Given user is at Pipeline Builder page
-    When user clicks Edit YAML button
+    When user clicks Edit YAML view
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/integration-tests/features/pipelines/create-from-builder-page.feature:110]:
```suggestion
    When user clicks YAML view
```

File: frontend/packages/dev-console/integration-tests/features/pipelines/create-from-builder-page.feature [hunk 2]
```diff
@@ -87,10 +87,27 @@ Scenario: Add Paramters to the pipeline in pipeline builder page : P-03-TC04
     And parameter details displayed in parameters section
 
 
+@regression, @manual
+Scenario: Switching from buider view to YAML view to see preserved data
+    Given user is at Pipeline Builder page
+    When user adds "git clone" task
+    And user clicks Edit YAML view
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/integration-tests/features/pipelines/create-from-builder-page.feature:94]:
```suggestion
    And user clicks YAML view
```

File: frontend/packages/dev-console/integration-tests/features/pipelines/create-from-builder-page.feature [hunk 3]
```diff
@@ -87,10 +87,27 @@ Scenario: Add Paramters to the pipeline in pipeline builder page : P-03-TC04
     And parameter details displayed in parameters section
 
 
+@regression, @manual
+Scenario: Switching from buider view to YAML view to see preserved data
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/integration-tests/features/pipelines/create-from-builder-page.feature:91]:
```suggestion
Scenario: Switching from Pipeline builder view to YAML view to see yaml data
```

File: frontend/packages/dev-console/integration-tests/features/pipelines/create-from-builder-page.feature [hunk 4]
```diff
@@ -87,10 +87,27 @@ Scenario: Add Paramters to the pipeline in pipeline builder page : P-03-TC04
...
+    Then user can see "git clone" task present in the YAML view with param values
+
+
+@regression, @manual
+Scenario: Switching from YAML view to builder view to see preserved data
+    Given user is at Pipeline YAML view page
+    When user adds "- name: git-clone" under tasks
+    And user adds "name: git-clone" and "kind: ClusterTask" under taskRef in tasks section 
+    And user clicks Pipeline Builder radio button 
+    Then user can see "git clone" tasks present in the Pipeline builder page
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/integration-tests/features/pipelines/create-from-builder-page.feature:104]:
```suggestion
    Then user can see "git clone" tasks present in the Pipeline builder view
```

File: frontend/packages/dev-console/integration-tests/features/pipelines/create-from-builder-page.feature [hunk 5]
```diff
@@ -87,10 +87,27 @@ Scenario: Add Paramters to the pipeline in pipeline builder page : P-03-TC04
...
+Scenario: Switching from buider view to YAML view to see preserved data
+    Given user is at Pipeline Builder page
+    When user adds "git clone" task
+    And user clicks Edit YAML view
+    Then user can see "git clone" task present in the YAML view with param values
+
+
+@regression, @manual
+Scenario: Switching from YAML view to builder view to see preserved data
+    Given user is at Pipeline YAML view page
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/integration-tests/features/pipelines/create-from-builder-page.feature:100]:
```suggestion
    Given user is at Pipeline YAML view
```

File: frontend/packages/dev-console/integration-tests/features/pipelines/pipelines-runs.feature [hunk 6]
```diff
@@ -206,3 +206,16 @@ Scenario Outline: Pipeline status display in topology side bar : P-05-TC02
 Examples:
 | pipeline_name          |
 | pipeline-with-resoruce |
+
+
+@regression, @manual
+Scenario: Tasks unaffected for existing pipeline runs when added a new task in the pipeline
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/integration-tests/features/pipelines/pipelines-runs.feature:212]:
```suggestion
Scenario: Editing a pipeline structure should not affect the previously executed pipeline runs.
```

File: frontend/packages/dev-console/integration-tests/features/pipelines/pipelines-runs.feature [hunk 7]
```diff
@@ -206,3 +206,16 @@ Scenario Outline: Pipeline status display in topology side bar : P-05-TC02
...
+    Given user has pipeline with two tasks
+    And user is at the Pipeline Details page
+    And pipeline run is present
+    When user clicks Edit Pipeline from Actions menu
+    And user adds a new task in series to the last task present
+    And user clicks on save 
+    And user goes to Pipeline Runs page
+    And user opens the given pipeline run
+    Then user can see two tasks in Pipeline Run details tab
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/integration-tests/features/pipelines/pipelines-runs.feature:221]: When user starts the pipeline
Then newly created pipeline run should contain the new pipeline graph
And existing pipeline runs should contain the old pipeline graph

File: frontend/packages/dev-console/integration-tests/features/pipelines/create-from-builder-page.feature (hunk 1 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/pipelines/create-from-builder-page.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/pipelines/create-from-builder-page.feature (hunk 2 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/pipelines/create-from-builder-page.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/pipelines/create-from-builder-page.feature (hunk 3 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/pipelines/create-from-builder-page.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/pipelines/create-from-builder-page.feature (hunk 4 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/pipelines/create-from-builder-page.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/pipelines/create-from-builder-page.feature (hunk 5 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/pipelines/create-from-builder-page.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/pipelines/pipelines-runs.feature (hunk 6 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/pipelines/pipelines-runs.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/pipelines/pipelines-runs.feature (hunk 7 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/pipelines/pipelines-runs.feature]: Updated

**Pull request merged**